tests/test_utils2.py

```diff
@@ -0,0 +1,74 @@
+import pytest
+from datetime import timezone
+
+from pystac.utils import (
+    rfc3339_str_to_datetime,
+    str_to_datetime,
+    now_to_rfc3339_str,
+    now_in_utc,
+)
+
+invalid_datetimes = [
+    "1985-04-12",  # date only
+    "1937-01-01T12:00:27.87+0100",  # invalid TZ format, no sep :
+    "37-01-01T12:00:27.87Z",  # invalid year, must be 4 digits
+    "1985-12-12T23:20:50.52",  # no TZ
+    "21985-12-12T23:20:50.52Z",  # year must be 4 digits
+    "1985-13-12T23:20:50.52Z",  # month > 12
+    "1985-12-32T23:20:50.52Z",  # day > 31
+    "1985-12-01T25:20:50.52Z",  # hour > 24
+    "1985-12-01T00:60:50.52Z",  # minute > 59
+    "1985-12-01T00:06:61.52Z",  # second > 60
+    "1985-04-12T23:20:50.Z",  # fractional sec . but no frac secs
+    "1985-04-12T23:20:50,Z",  # fractional sec , but no frac secs
+    "1990-12-31T23:59:61Z",  # second > 60 w/o fractional seconds
+    "1985-04-12T23:20:50,52Z",  # comma as frac sec sep disallowed in RFC3339
+]
+
+valid_datetimes = [
+    "1985-04-12T23:20:50.52Z",
+    "1996-12-19T16:39:57-00:00",
+    "1996-12-19T16:39:57+00:00",
+    "1996-12-19T16:39:57-08:00",
+    "1996-12-19T16:39:57+08:00",
+    "1937-01-01T12:00:27.87+01:00",
+    "1985-04-12T23:20:50.52Z",
+    "1937-01-01T12:00:27.8710+01:00",
+    "1937-01-01T12:00:27.8+01:00",
+    "1937-01-01T12:00:27.8Z",
+    "2020-07-23T00:00:00.000+03:00",
+    "2020-07-23T00:00:00+03:00",
+    "1985-04-12t23:20:50.000z",
+    "2020-07-23T00:00:00Z",
+    "2020-07-23T00:00:00.0Z",
+    "2020-07-23T00:00:00.01Z",
+    "2020-07-23T00:00:00.012Z",
+    "2020-07-23T00:00:00.0123Z",
+    "2020-07-23T00:00:00.01234Z",
+    "2020-07-23T00:00:00.012345Z",
+    "2020-07-23T00:00:00.0123456Z",
+    "2020-07-23T00:00:00.01234567Z",
+    "2020-07-23T00:00:00.012345678Z",
+]
+
+
+@pytest.mark.parametrize("test_input", invalid_datetimes)
+def test_parse_invalid_str_to_datetime(test_input: str) -> None:
+    with pytest.raises(ValueError):
+        rfc3339_str_to_datetime(test_input)
+
+
+@pytest.mark.parametrize("test_input", valid_datetimes)
+def test_parse_valid_str_to_datetime(test_input: str) -> None:
+    assert str_to_datetime(test_input)
+    assert rfc3339_str_to_datetime(test_input)
+
+
+def test_now_functions() -> None:
+    now1 = now_in_utc()
+    now2 = now_in_utc()
+
+    assert now1 < now2
+    assert now1.tzinfo == timezone.utc
+
+    rfc3339_str_to_datetime(now_to_rfc3339_str())
```

Comment:
Could be cool to break this out into its own micropackage to assess various libraries, do you have anything like that already @philvarner? If not, I might have a couple of cycles...